Fix stripping of python libraries

plettich opened the pull request:
- Remove the stripping of libraries in post install macros.
- Remove several build dependencies, they are not needed any more.
- During setup of the virtualenv, first install the pinned requirements
  from github for this version, then install privacyIDEA from pypi.
- Split up building and signing of the RPM packages.

plettich committed:
Fix stripping of python libraries

- Remove the stripping of libraries in post install macros.
- Remove several build dependencies, they are not needed any more.
- During setup of the virtualenv, first install the pinned requirements
  from github for this version, then install privacyIDEA from pypi.
- Split up building and signing of the RPM packages.

cornelinux commented on Makefile:
```diff
@@ -6,20 +6,26 @@ PI_VERSION=${VERSION}
-buildrpm:
-	PI_VERSION=${PI_VERSION} rpmbuild --define "_topdir `pwd`" --sign -ba SPECS/privacyidea.spec
-	PI_VERSION=${PI_VERSION} rpmbuild --define "_topdir `pwd`" --sign -ba SPECS/privacyidea-server.spec
+buildrpm: buildpi buildradius buildoracle
```
The problem is that currently the documentation tells you to do a ``buildrpm``.
But: The buildoracle probably will never be run again in the next 2 years ;-)

So the build process changes a lot. We would have to change the docs, too.
I am not sure, if changing the buildrpm directive is a good idea.

plettich replied:
will change that

plettich replied:
fixed in ffdb7b0

cornelinux commented on Makefile:
```diff
@@ -6,20 +6,26 @@ PI_VERSION=${VERSION}
+	rpmbuild --define "_topdir `pwd`" -ba SPECS/privacyidea-cx-oracle.spec
+
+signrpm: buildrpm
+	find RPMS/ -name *.rpm -exec 'rpmsign' '--addsign' '{}' ';'
```
I am lazy! I do not like to run too many make directives.

(see below...)

plettich replied:
see below :-)

cornelinux commented on Makefile:
```diff
@@ -6,20 +6,26 @@ PI_VERSION=${VERSION}
+signrpm: buildrpm
+	find RPMS/ -name *.rpm -exec 'rpmsign' '--addsign' '{}' ';'
 
 fill-release-repo:
```
In fill-release-repo and fill-devel-repo we could run

    make sign

as the first step. Since we can not (or should not) add unsigned RPMs into the repository anyways!

plettich replied:
makes sense

plettich replied:
fixed in ffdb7b0

cornelinux commented on SPECS/privacyidea.spec:
```diff
@@ -18,7 +19,7 @@ Packager:       Cornelius Kölbel <cornelius.koelbel@netknights.it>
 BuildArch:      x86_64
 AutoReqProv:	no
 
-BuildRequires: libxml2-devel, freetype-devel, python-devel, libxslt-devel, zlib-devel, openssl-devel, python-virtualenv, gcc, createrepo
```
Are you sure, we do not need some of the depencies? (Although they might be installed by default)

plettich replied:
Well, i tested building the RPMs without these packages installed, it did seem to work... We removed a few dependencies that needed building some libs.

plettich replied:
tested and works. We really don't need the devel build dependencies anymore.

cornelinux commented on SPECS/privacyidea.spec:
```diff
@@ -40,9 +41,9 @@ rm -fr /opt/privacyidea
 virtualenv /opt/privacyidea
 source /opt/privacyidea/bin/activate
 pip install --upgrade pip
+pip install -r https://raw.githubusercontent.com/privacyidea/privacyidea/v%{version}/requirements.txt
```
There could be corner cases where the git repo contains another requirement in the tags than the python package index.
Should we install the privacyidea from github completely? We do this with the ubuntu packages.
I am not sure due to which problems we decided to install from pypi...

plettich replied:
I'll test installing it from github.

plettich replied:
fixed in ffdb7b0

plettich committed:
Fix remarks from review

- more meaningful error message if VERSION not provided
- buildrpm target builds only the privacyidea and privacyidea-server
  packages
- add signrpm target as prerequisites for fill-release-repo and
  fill-devel-repo targets
- add a mkdir in case the repository directory does not exist
- the privacyidea package now pulls requirements and code from a github tag
- update build requirements, we don't need the devel packages anymore

cornelinux commented:
The orignial build process fetches the code from pypi.
Your changes fetches the code from github. We could do this, but please also change the README.md so that the builder knows, which version from where will be built.

So when this is fixed, we probably should finally merge this.

plettich commented:
This PR is obsolete, please look at #8
